Content.Client/RCD/RCDConstructionGhostSystem.cs

```diff
@@ -0,0 +1,253 @@
+using Content.Client.Gameplay;
+using Content.Shared.Hands.Components;
+using Content.Shared.RCD;
+using Content.Shared.RCD.Components;
+using Content.Shared.RCD.Systems;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Input;
+using Robust.Client.Player;
+using Robust.Client.State;
+using Robust.Shared.Input;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.RCD;
+
+public sealed class RCDConstructionGhostSystem : EntitySystem
+{
+    [Dependency] private readonly IInputManager _inputManager = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IEyeManager _eyeManager = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly IPrototypeManager _protoManager = default!;
+    [Dependency] private readonly IStateManager _stateManager = default!;
+    [Dependency] private readonly SharedMapSystem _mapSystem = default!;
+    [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly RCDSystem _rcdSystem = default!;
+
+    private EntityUid? _constructionGhost = null;
+    private ProtoId<RCDPrototype>? _constructionPrototype = null;
+    private RCDPrototype? _cachedPrototype = null;
+    private bool _isKeyBindActive = false;
+
+    private readonly BoundKeyFunction _keyfunction = EngineKeyFunctions.EditorRotateObject;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        // Determine if player is carrying an RCD in their active hand
+        var player = _playerManager.LocalSession?.AttachedEntity;
+
+        if (!TryComp<HandsComponent>(player, out var hands))
+            return;
+
+        var uid = hands.ActiveHand?.HeldEntity;
+        var hasRCD = TryComp<RCDComponent>(uid, out var rcd);
+
+        // Rotate the construction ghost if the keybind was actived this frame
+        if (WasKeybindActivatedThisFrame(player.Value) && hasRCD)
+            RotateRCDConstructionGhost(uid!.Value, rcd!);
+
+        // Delete the construction ghost if its no longer needed
+        if (!hasRCD ||
+            !EntityManager.TryGetComponent<InputComponent>(player, out var input) ||
+            _inputManager.Contexts.ActiveContext.Name != input.ContextName)
+        {
+            DeleteConstructionGhost();
+            return;
+        }
+
+        if (_constructionPrototype != rcd!.ProtoId)
```

Comment:
doing null suppression here is really not necessary. just do a null check just in case.